cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -277,13 +279,18 @@ public synchronized RunId run(ProgramId programId, Map<String, String> overrides
    */
   public synchronized RunId runInternal(ProgramId programId, Map<String, String> userArgs,
                                         Map<String, String> sysArgs,
-                                        boolean debug) throws NotFoundException, IOException {
+                                        boolean debug) throws NotFoundException, IOException, BadRequestException {
     LOG.info("{} tries to run {} Program {}", authenticationContext.getPrincipal().getName(), programId.getType(),
              programId.getProgram());
-    String scopedProfile = userArgs.get(SystemArguments.PROFILE_NAME);
-    ProfileId profileId = scopedProfile == null ?
-      ProfileId.DEFAULT : ProfileId.fromScopedName(programId.getNamespaceId(), scopedProfile);
-    Profile profile = profileStore.getProfile(profileId);
+    ProfileId profileId =
+      SystemArguments.getProfileIdFromArgs(programId.getNamespaceId(), userArgs).orElse(ProfileId.DEFAULT);
+    ProfileMeta profileMeta = profileStore.getProfileMeta(profileId);
+    if (profileMeta.getStatus() == ProfileStatus.DISABLED) {
+      throw new BadRequestException(String.format("Profile %s in namespace %s is disabled. So it cannot be " +
```

Comment:
I think it's better to use a ConflictException for this situation. When somebody tries to start a program that is already running and that doesn't allow concurrent runs, we throw a ConflictException. I think this is a similar scenario, where the request is valid, but is just not allowed in the current state.

Reply:
fixed